### DISCUSSION TITLE
Interrupts for physical buttons

### DISCUSSION THREAD

**Discussion by Username-cannot-be-blank** [PlummersSoftwareLLC/NightDriverStrip#8] [Ideas]
I've been writing something similar to this for my own project, although not nearly as well engineered as this. 
I'm designing mine to work without WiFi for a couple of reasons, the main one being I want the option to be able to mount a device to an ATV where the internet connection may be spotty or non-existent. 
To that end, I have built a control box with physical buttons that respond to interrupt requests when pressed (I'm using pull up resistors to keep the pins high and the switch pulls them low)
One button controls the effect, another controls then color palette, another the brightness, and so forth. 
I think it should be possible to incorporate these controls, but I'm not sure where the appropriate place to put such functionality would be. 
Anyone have any ideas?

**Comment by davepl** [comment]
Yes, take a look at giInfoPage.  It controls which page of info is displayed on the main display, and its completely denounced in hardware using interrupts.  Right now it is only set up for the main button of the M5, but its in there.  It uses the Debounce library which seems to work well enough!
